lib/utils.py

```diff
@@ -110,8 +114,11 @@ def get_current_time() -> datetime:
         # Set a longer timeout to make the request more reliable
         response = c.request(NTP_SERVER, version=3, timeout=10)
     except (socket.gaierror, ntplib.NTPException):
-        logger.debug("Error requesting time from NTP server. Using local time")
-        return datetime.now(timezone.utc).replace(tzinfo=None)
+        try:
```

Comment:
I think it would be best to split this into another PR if it isn't relevant to the check-in changes. What are your thoughts? Is it directly related or just another separate change?